### PR TITLE
Feature/journal and OU indexing

### PR DESCRIPTION
**Pull request by ivanmrsulja**
## Description
Added indexing of journals and OUs. Added some supplementary methods for fetching proceedings and publications for an event.

## How should this be tested?
Unit and integrtion tests are implemented for every newly added functionality. You can test each new endpoint using client mocking tool e.g. Postman.

## Checklist

-   [x] I have tested these changes locally
-   [x] I have added tests that cover these changes (if applicable)
-   [x] I have updated the documentation (if applicable)
-   [x] I have added or updated the necessary comments in the code

### CODE REVIEW

File: src/main/java/rs/teslaris/core/controller/ProceedingsPublicationController.java [hunk 1]
```diff
@@ -24,13 +29,25 @@ public class ProceedingsPublicationController {
...
     @GetMapping("/{publicationId}")
     public ProceedingsPublicationDTO readProceedingsPublication(
         @PathVariable Integer publicationId) {
         return proceedingsPublicationService.readProceedingsPublicationById(publicationId);
     }
 
+    @GetMapping("/for-event/{eventId}")
+    public List<ProceedingsPublicationResponseDTO> readProceedingsPublicationsForEvent(
+        @PathVariable Integer eventId, @RequestHeader("Authorization") String bearerToken) {
+        return proceedingsPublicationService.findProceedingsForEvent(eventId,
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/controller/ProceedingsPublicationController.java:46]: This should return conference articles (ProceedingsPublication) from a certain event (conference) which author is loggedIn researcher, right? If that is the case, not sure the name of method is accurate. Does this return proceedings or proceedings publications?

File: src/main/java/rs/teslaris/core/controller/ProceedingsPublicationController.java [hunk 2]
```diff
@@ -24,13 +29,25 @@ public class ProceedingsPublicationController {
...
+    private final UserService userService;
+
 
     @GetMapping("/{publicationId}")
     public ProceedingsPublicationDTO readProceedingsPublication(
         @PathVariable Integer publicationId) {
         return proceedingsPublicationService.readProceedingsPublicationById(publicationId);
     }
 
+    @GetMapping("/for-event/{eventId}")
```
Comment (chenejac) [inline src/main/java/rs/teslaris/core/controller/ProceedingsPublicationController.java:43]: for the JournalPublicationController this endpoint is /my-publications/{journalId}. Can we align this? Maybe we can use api/proceedings-publication/event/{eventId}/my-publications and /api/journal-publication/journal/{journalId}/my-publications. Besides those endpoints for my-publications, we will need also api/proceedings-publication/event/{eventId} and /api/journal-publication/journal/{journalId}, which will return all publications from event or conference (for the event and journal profile page).

**Review by chenejac — CHANGES_REQUESTED** [review]
@ivanmrsulja please check my two comments. Thanks

File: src/main/java/rs/teslaris/core/controller/ProceedingsPublicationController.java (hunk 1 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/controller/ProceedingsPublicationController.java:46]: This returns ProceedingsPublications.

File: src/main/java/rs/teslaris/core/controller/ProceedingsPublicationController.java (hunk 2 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/controller/ProceedingsPublicationController.java:43]: Ok, I will align the paths as you suggested and add missing methods.

File: src/main/java/rs/teslaris/core/controller/ProceedingsPublicationController.java (hunk 1 above)
Comment (ivanmrsulja) [reply src/main/java/rs/teslaris/core/controller/ProceedingsPublicationController.java:46]: I renamed it to `readMyProceedingsPublicationsForEvent()` Is that ok?

**Pull request merged**